### PR TITLE
Jade/ecb violations

### PR DESCRIPTION
**Pull request by 0xStarcat**
Adds a brief "How to add a new dataset" guide

adds the NYC Open Data ECB Violations dataset and integration test

### CODE REVIEW

File: src/nycdb/sql/dob_violations.sql [hunk 1]
```diff
@@ -1,3 +1,4 @@
 CREATE INDEX dob_violations_bbl_idx on dob_violations (bbl);
 CREATE UNIQUE INDEX dob_violations_isndobbisviol_idx on dob_violations (isndobbisviol);
 CREATE INDEX dob_violations_violationtypecode on dob_violations (violationtypecode);
+CREATE UNIQUE INDEX dob_violations_ecb_number_idx on ecb_violations (EcbNumber);
```
Comment (0xStarcat) [inline src/nycdb/sql/dob_violations.sql]: some indexing on the fields linking `dob_violations` with `ecb_violations`

**Comment by aepyornis** [comment]
hi @0xStarcat ! Thank you! This is awesome.

You might try rebasing with master and removing the changes to date function because I think the issue has been fixed with https://github.com/aepyornis/nyc-db/commit/a800863fbe7f635bfb7346cac34ac6765a444354 

The guide is excellent and very helpful. I think we should add a section about  how to writing tests  for the new dataset (and ideally include one to for this dataset as well). In my opinion, having an integration test for each dataset is really beneficial.

File: src/tests/integration/test_nycdb.py [hunk 2]
```diff
@@ -14,10 +14,10 @@
 data_dir = os.path.join(os.path.dirname(__file__), 'data')
 
 ARGS = SimpleNamespace(
-    user=os.environ.get('NYCDB_TEST_POSTGRES_USER', 'postgres'),
-    password=os.environ.get('NYCDB_TEST_POSTGRES_PASSWORD', 'password'),
+    user=os.environ.get('NYCDB_TEST_POSTGRES_USER', 'nycdb'),
```
Comment (0xStarcat) [inline src/tests/integration/test_nycdb.py]: i changed this fallback values to match the environment variables set in the `Dockerfile` and now the tests work

File: src/tests/integration/test_nycdb.py [hunk 3]
```diff
@@ -93,13 +93,21 @@ def has_one_row(conn, query):
 
 
 def table_columns(conn, table_name):
-    sql = "SELECT column_name FROM information_schema.columns WHERE table_schema = 'public' AND table_name = '{}'".format(table_name)
+    sql = "SELECT column_name FROM information_schema.columns WHERE table_schema = 'public' AND table_name = '{}'".format(
+        table_name)
```
Comment (0xStarcat) [inline src/tests/integration/test_nycdb.py]: linter

**Comment by 0xStarcat** [comment]
I changed the DB name, user name, and password in the test file to match those used by the dockerfile. Tests are probably failing in travis because the travis variables don't match. Would you be able to change the travis variables to match the dockerfile/test variables? I think it may make sense to have them all be the same.

**Comment by aepyornis** [comment]
Hey @0xStarcat  

I fixed the the travis issue here -- e5ea46df9f90c595b7da6b1cc13daeb3947b88a7. those db connection variables actually lived in ` .travis.yml `  :)

Merged!

**Comment by 0xStarcat** [comment]
> Hey @0xStarcat
> 
> I fixed the travis issue here -- [e5ea46d](https://github.com/aepyornis/nyc-db/commit/e5ea46df9f90c595b7da6b1cc13daeb3947b88a7). those db connection variables were actually hiding in `.travis.yml` :)
> 
> Merged!

yipee!

**Pull request merged**